source/common/http/http1/codec_impl.cc

```diff
@@ -136,6 +136,7 @@ void StreamEncoderImpl::encodeHeadersBase(const RequestOrResponseHeaderMap& head
 
   if (headers.ContentLength()) {
     saw_content_length = true;
+    content_length_ = !disable_content_length_;
```

Comment:
so we're not really doing framing by half close here if we didn't see T-E chunked or content length, we're force doing it for connect requests.

I think we should either
1) just set half close for connect requests, and half-close for connect requests when we hit end stream
2) really do half close any time we don't have content length or te-chunked.

right now it looks like we're mixing and matching (or I'm misreading the code - please correct me if I'm wrong!)

Reply:
This was just to ignore the `content_length_` flag in the scope of a CONNECT request, not sure it was a good idea though :-)
I took the first approach you suggested.